cmd/thanos/receive.go

```diff
@@ -429,7 +429,7 @@ func runReceive(
 					WriteableStoreServer: webHandler,
 				}
 
-				s = grpcserver.NewReadWrite(logger, &receive.UnRegisterer{Registerer: reg}, tracer, comp, grpcProbe, rw,
+				s = grpcserver.NewReadWrite(logger, reg, tracer, comp, grpcProbe, rw,
```

Comment:
have you tested this change? if you trigger a hashring change and a new grpc server is created and registers its metrics against the registerer, then this will panic

Reply:
yes correct I overlooked this.